### DISCUSSION TITLE
Option: Local Identities ➡ Global Identities?

### DISCUSSION THREAD

**Discussion by devmount** [devmount/third-stats#252] [Changes] [answered]
Currently it's possible to define identities in the ThirdStats options, that affect local accounts only:
![image](https://user-images.githubusercontent.com/5441654/108022211-6e7ecc00-7020-11eb-80e0-72fe06c31984.png)

 I implemented this because Thunderbird doesn't offer a possibility to define identities for local accounts. But for some users it's not clear, that the identities configured in the ThirdStats Options won't apply all existing accounts and wonder, why emails are not properly recognized as sent or self message.

I see the following possibilities:

1. Leave it as is, the name "**Local** Identities" clearly implies that this is only for local accounts.
2. Rename it to "Global Identities" and take the union of all identities configured in Thunderbird and ThirdStats
3. Rename it to "Identities" and only take these identities configured in ThirdStats to prevent confusion about identities configured somewhere else in Thunderbird

What do you all think about that? Which of the 3 possibilities do you think is the best here?

(for awareness: @Eagle3386 @koobs @rfwolff, @ramonrrp)

**Comment by Eagle3386** [comment]
Thanks for the opportunity, @devmount!

Regarding your question, I'd say either go with option 1 or 3 as I strongly suspect more confusion to arise if going with option 2.

If either option shall be chosen, I'd rather stick with the third than the first. However, I'd like to suggest to consider a mix of both, i. e. improve the explanation beneath `Local identities` _and_ renaming the option itself - at least bit.

**Reply by devmount** [reply]
Wow, now that was a quick response 👏🏻 Thank you @Eagle3386 for commenting this fast.

The problem I see with option 3 is that people having a lot of identities configured in Thunderbird need to configure them again in ThirdStats, which is annoying double configuration I wanted to prevent. I like your idea of mixing both. Thinking about it I also tend to option 1 with an improved description including a hint, how "normal identities" are configured in Thunderbird.

Any other opinions?

**Comment by thetrial** [comment]
I recently fed in tons of identities into TB – just to see correct stats. Though I don’t use those identities (anymore). But I have a large archive for nearly 30 years, and you can imagine … there were several accounts in these times. So to put all these into this plugin would be nice. I tried it before because I misunderstood the point. So afterwards I started to add all bypassed accounts as identities.

**Reply by devmount** [reply, marked as answer]
I see. Yes I understand that in case of several "deprecated identities" it would be much more convenient to simply add them in the add-on options.

Maybe it's best to rename that option to "Identities", merge those identities with existing ones configured in Thunderbird, use them for all accounts (local and non local) and provide a list of all identities, that are already configured in Thunderbird. This way, you have an overview of all existing identities as well as adding identities only in ThirdStats or only in Thunderbird. It also would reduce confusion since all identities are visible in the ThirdStats options.

**Reply by thetrial** [reply]
I guess, this could be the easiest way, yes.
